src/storage/exec/GetPropNode.h

```diff
@@ -56,35 +63,58 @@ class GetTagPropNode : public QueryNode<VertexID> {
     auto isIntId = context_->isIntId();
     for (auto* tagNode : tagNodes_) {
       ret = tagNode->collectTagPropsIfValid(
-          [&row](const std::vector<PropContext>* props) -> nebula::cpp2::ErrorCode {
+          [&row, tagNode, this](const std::vector<PropContext>* props) -> nebula::cpp2::ErrorCode {
             for (const auto& prop : *props) {
               if (prop.returned_) {
                 row.emplace_back(Value());
               }
+              if (prop.filtered_ && expCtx_ != nullptr) {
+                expCtx_->setTagProp(tagNode->getTagName(), prop.name_, Value());
+              }
             }
             return nebula::cpp2::ErrorCode::SUCCEEDED;
           },
-          [&row, vIdLen, isIntId](
+          [&row, vIdLen, isIntId, tagNode, this](
               folly::StringPiece key,
               RowReader* reader,
               const std::vector<PropContext>* props) -> nebula::cpp2::ErrorCode {
-            if (!QueryUtils::collectVertexProps(key, vIdLen, isIntId, reader, props, row).ok()) {
-              return nebula::cpp2::ErrorCode::E_TAG_PROP_NOT_FOUND;
+            for (const auto& prop : *props) {
```

Comment:
Maybe unify the code in `HashJoinNode` into a function?

Reply:
Done.